airflow/executors/kubernetes_executor.py

```diff
@@ -246,7 +249,9 @@ def process_status(
             self.log.info("Event: %s Succeeded", pod_name)
             self.watcher_queue.put((pod_name, namespace, None, annotations, resource_version))
         elif status == "Running":
-            if event["type"] == "DELETED":
+            # deletion_timestamp is set by kube server when a graceful deletion is requested.
+            # since the pod is in intermediate state so set TI status failed
+            if pod.metadata.deletion_timestamp is not None:
```

Comment:
```suggestion
            if event["type"] == "DELETED" and pod.metadata.deletion_timestamp is not None:
```
Not sure if the `is not None` is necessary